crates/accelerate/src/quantum_circuit/circuit_instruction.rs

```diff
@@ -70,26 +69,27 @@ impl CircuitInstruction {
     pub fn new(
         py: Python<'_>,
         operation: PyObject,
-        qubits: Option<&PyAny>,
-        clbits: Option<&PyAny>,
+        qubits: Option<&Bound<PyAny>>,
+        clbits: Option<&Bound<PyAny>>,
     ) -> PyResult<Self> {
```

Comment:
Idle curiosity: do you know why we need `&Bound` and can't use an owned `Bound`, like we can accept an owned `Py`?

Reply:
I believe it's just fine to accept an owned `Bound`. I mostly just followed the warning's guidance on the old GIL ref, which suggested this be ported by making it a bound reference (and, I saw @mtreinish do this in the Rustworkx migration PR).

I'm not sure exactly how the generated PyO3 code looks coming in to / out of this method, but perhaps accepting an owned bound could require incrementing a ref count if PyO3 doesn't want to move the bounds it extracted in (cloning a bound increments the ref count, IIRC).

For our purposes, when calling `pymethods` from Rust code, it seems better to do so via reference for the same reason.

Reply:
I had a bit of a look, but not to the point that I can be sure I'm right, more just I tried to get a feeling.  Roughly, I _think_ that `Bound` would be fine and PyO3 would have a `Bound` that it can give us ownership without issues _but_ it still suggests using `&Bound`, because `&Bound` is the better standard for if you also want to use the `pyfunction` from within Rust, without going via the PyO3 trampoline - there's no reason the intra-Rust call should need an owned `Bound`.  I think the refcount update should in theory be very cheap (since the existence of a `Bound` at all implies we already hold the GIL), but it's still avoidable using Rust's ownership model.

I might be making that up completely, but it at least makes some logical sense to me.

Reply:
Sounds right, and I think we're saying the same thing 😄.

Reply:
oh, so we are - I totally overlooked your very last sentence there when I was writing the response - I got too wrapped up in looking at pyo3 myself lol